front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/ChartWidgets/VisitorsWidget/VisitorsCard/Wide.tsx

```diff
@@ -34,8 +42,20 @@ const Wide = ({
   stats,
   timeSeries,
   hideStatistics,
+  ariaLabel,
+  description,
 }: Props) => {
   const previousDays = getDaysInRange(startAt, endAt);
+  const localize = useLocalize();
+  const { formatMessage } = useIntl();
+
+  // Generate unique IDs for ARIA attributes
+  const chartId = React.useId();
+  const descriptionId = `${chartId}-description`;
+
+  // Get localized values
+  const localizedAriaLabel = ariaLabel ? localize(ariaLabel) : undefined;
+  const localizedDescription = description ? localize(description) : undefined;
 
```

Comment:
Rather than repeating this in many components:

```
  // Get localized values
  const localizedAriaLabel = ariaLabel ? localize(ariaLabel) : undefined;
  const localizedDescription = description ? localize(description) : undefined;
```
and
```
    {/* description for screen readers */}
      {localizedDescription && (
        <Text color="grey700" fontSize="s" id={descriptionId}>
          {formatMessage(messages.description)} {localizedDescription}
        </Text>
      )}
```

Maybe we can make a re-usable component in the _shared folder instead? Called `<ChartDescription/>` or something? If it accepts the `chartId`, `ariaLabel`, `descriptionId ` & `description`, I think we could easily re-use it in all the different chart components :relaxed: